test/src/module/storage/azureTest.c

```diff
@@ -0,0 +1,783 @@
+                        "            <Properties/>"
+                        "        </Blob>"
+                        "        <BlobPrefix>"
+                        "            <Name>path/to/path1/</Name>"
+                        "        </BlobPrefix>"
+                        "    </Blobs>"
+                        "    <NextMarker>ueGcxLPRx1Tr</NextMarker>"
+                        "</EnumerationResults>");
+
+                        // "<ListBucketResult xmlns=\"http://s3.amazonaws.com/doc/2006-03-01/\">"
+                        // "    <NextContinuationToken>1ueGcxLPRx1Tr/XYExHnhbYLgveDs2J/wm36Hy4vbOwM=</NextContinuationToken>"
+                        // "    <Contents>"
+                        // "        <Key>path/to/test1.txt</Key>"
+                        // "    </Contents>"
+                        // "    <Contents>"
+                        // "        <Key>path/to/test2.txt</Key>"
+                        // "    </Contents>"
+                        // "   <CommonPrefixes>"
+                        // "       <Prefix>path/to/path1/</Prefix>"
+                        // "   </CommonPrefixes>"
+                        // "</ListBucketResult>");
+
+                testRequestP(HTTP_VERB_GET, "?comp=list&delimiter=%2F&marker=ueGcxLPRx1Tr&prefix=path%2Fto%2F&restype=container");
+                testResponseP(
+                    .content =
+                        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
+                        "<EnumerationResults>"
+                        "    <Blobs>"
+                        "        <Blob>"
+                        "            <Name>path/to/test3.txt</Name>"
+                        "            <Properties/>"
+                        "        </Blob>"
+                        "        <BlobPrefix>"
+                        "            <Name>path/to/path2/</Name>"
+                        "        </BlobPrefix>"
+                        "    </Blobs>"
+                        "    <NextMarker/>"
+                        "</EnumerationResults>");
+
+                callbackData.content = strNew("");
+
+                TEST_RESULT_VOID(
+                    storageInfoListP(
+                        storage, strNew("/path/to"), hrnStorageInfoListCallback, &callbackData, .level = storageInfoLevelExists),
+                    "list");
+                TEST_RESULT_STR_Z(
+                    callbackData.content,
+                    "path1 {}\n"
+                    "test1.txt {}\n"
+                    "test2.txt {}\n"
+                    "path2 {}\n"
+                    "test3.txt {}\n",
+                    "check");
+
+                // -----------------------------------------------------------------------------------------------------------------
+                TEST_TITLE("list files with expression");
+
+                testRequestP(HTTP_VERB_GET, "?comp=list&delimiter=%2F&prefix=path%2Fto%2Ftest&restype=container");
```

Comment:
You indicated "reason the test breaks is that the URI expected at line 637 changes since the prefix no longer includes `/test`"  So if I want the expression to be `strNew("test(1|3)")` (without the caret) then what would the URI be? I tried it without `/test `and it still failed.

Reply:
The final `/` is still required because we are asking for all files in the `path/to` path. Without the / we would be looking for files that begin with `to` in the `path` path.

So the correct URI when the carat is removed from the expression is: `?comp=list&delimiter=%2F&prefix=path%2Fto%2F&restype=container`.